### PR TITLE
Add support for finding updated 7-Zip versions on Linux

### PR DESCRIPTION
**Pull request by Dead2**
The old linux port of 7-Zip named p7zip has not been maintained since 2016, and a few updates has come out for 7zip since.
In 2022, 7-Zip released its own linux port, but uses different commands for its port to avoid colliding with p7zip.

This PR adds support for detecting the `7zz` and ` 7zzs` commands.

A lot has happened since 16.01, ref: https://www.7-zip.org/history.txt
For example, a new `ARM64` filter has been introduced for LZMA, I am guessing archives compressed using this will fail to unpack using older versions.

My own benchmarks on x86_64:

p7zip 16.01:
```
# 7za b
7-Zip (a) [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.UTF-8,Utf16=on,HugeFiles=on,64 bits,8 CPUs Intel(R) Xeon(R) CPU E3-1245 v6 @ 3.70GHz (906E9),ASM,AES-NI)

Intel(R) Xeon(R) CPU E3-1245 v6 @ 3.70GHz (906E9)
CPU Freq: - - - - - - - - -

RAM size:   64206 MB,  # CPU hardware threads:   8
RAM usage:   1765 MB,  # Benchmark threads:      8

                       Compressing  |                  Decompressing
Dict     Speed Usage    R/U Rating  |      Speed Usage    R/U Rating
         KiB/s     %   MIPS   MIPS  |      KiB/s     %   MIPS   MIPS

22:      24370   652   3638  23708  |     262333   798   2803  22376
```

7-Zip 23.01:
```
# 7zz b
7-Zip (z) 23.01 (x64) : Copyright (c) 1999-2023 Igor Pavlov : 2023-06-20
 64-bit locale=en_US.UTF-8 Threads:8 OPEN_MAX:1024, ASM

Compiler: 9.4.0 GCC 9.4.0: SSE2
Linux : 6.1.27-1.el9.x86_64 : #1 SMP Fri May  5 12:31:25 CEST 2023 : x86_64
PageSize:4KB THP:madvise hwcap:2 hwcap2:2
Intel(R) Xeon(R) CPU E3-1245 v6 @ 3.70GHz (906E9)

1T CPU Freq (MHz):  4042  4069  4050  4047  4065  4051  4060
4T CPU Freq (MHz): 399% 3895   398% 3897

RAM size:   64206 MB,  # CPU hardware threads:   8
RAM usage:   1779 MB,  # Benchmark threads:      8

                       Compressing  |                  Decompressing
Dict     Speed Usage    R/U Rating  |      Speed Usage    R/U Rating
         KiB/s     %   MIPS   MIPS  |      KiB/s     %   MIPS   MIPS

22:      29936   629   4629  29122  |     410380   797   4391  34994
```

Decompression of LZMA went from 262MB/s to 410MB/s, a whopping 56% speedup.
According to changelist, ARM should have an even bigger speedup.

### CODE REVIEW

**Comment by sanderjo** [comment]
Let me check on my Ubuntu.

TLDR: yes, looks like described by @Dead2 

```
$ 7z | head -4

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.UTF-8,Utf16=on,HugeFiles=on,64 bits,4 CPUs 11th Gen Intel(R) Core(TM) i3-1115G4 @ 3.00GHz (806C1),ASM,AES-NI)

```
Looks old indeed

```
$ sudo apt install 7zip -y

$ 7zz | head

7-Zip (z) 21.07 (x64) : Copyright (c) 1999-2021 Igor Pavlov : 2021-12-26
 64-bit locale=en_US.UTF-8 Threads:4

```
Ah, so `7zz` is what Linux users should use? That is: if they use 7zip.

**Comment by Safihre** [comment]
Great 👍
